kernel: add Aquantia AQtion USB-to-5GbE adapters

mj22226 opened the pull request:
This add support for USB-to-Ethernet Aquantia AQtion
5/2.5GbE adapters based on the AQC111U controllers.

Run-tested: x86
Adapter-tested: Sabrent NT-SS5G

lucize commented:
this driver is compatible only with x86 ?

mj22226 commented:
> this driver is compatible only with x86 ?

I had only tested on x86, with heavy loads it draw about 0,5 amp from the USB port and run a little hot, I think is not safe to use it in all targets.

bobafetthotmail commented:
> 
> I had only tested on x86, with heavy loads it draw about 0,5 amp from the USB port and run a little hot, I think is not safe to use it in all targets.

USB 2.0 port is rated for 0.5 amp and USB 3.0 port is rated for 0.9 amp, it is OK, especially because it's crazy to use this on a USB 2.0 port.
USB hard drives need similar amount of power.

I think you can remove the DEPENDS:=@TARGET_x86

If you don't have devices with other architectures, you can compile some firmware images for devices using a different architecture, just to check that it compiles without errors.
Try with a device in **mvebu** and one in **ipq806x**, they are powerful devices and usually have USB 3.0.

mj22226 commented:
> Try with a device in **mvebu** and one in **ipq806x**, they are powerful devices and usually have USB 3.0.

I tried on mvebu - wrt3200 and is not working.
On ipq806x - R7800 works but network speed is slow.
Server running on the r7800 and client on the PC:
```
root@OpenWrt:/# ethtool eth2 | grep -i speed
        Speed: 5000Mb/s
root@OpenWrt:/# iperf3 -s
-----------------------------------------------------------
Server listening on 5201
-----------------------------------------------------------
Accepted connection from 192.168.1.199, port 56862
[  5] local 192.168.1.1 port 5201 connected to 192.168.1.199 port 56864
[ ID] Interval           Transfer     Bitrate
[  5]   0.00-1.00   sec  75.8 MBytes   635 Mbits/sec                  
[  5]   1.00-2.00   sec  78.5 MBytes   658 Mbits/sec                  
[  5]   2.00-3.00   sec  77.7 MBytes   652 Mbits/sec                  
[  5]   3.00-4.00   sec  77.7 MBytes   652 Mbits/sec                  
[  5]   4.00-5.00   sec  77.7 MBytes   651 Mbits/sec                  
[  5]   5.00-6.00   sec  78.0 MBytes   655 Mbits/sec                  
[  5]   6.00-7.00   sec  78.1 MBytes   655 Mbits/sec                  
[  5]   7.00-8.00   sec  77.7 MBytes   652 Mbits/sec                  
[  5]   8.00-9.00   sec  77.7 MBytes   652 Mbits/sec                  
[  5]   9.00-10.00  sec  77.8 MBytes   652 Mbits/sec                  
[  5]  10.00-10.01  sec  1000 KBytes   623 Mbits/sec                  
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bitrate
[  5]   0.00-10.01  sec   778 MBytes   651 Mbits/sec                  receiver
-----------------------------------------------------------
```
Server running on the PC and client on the r7800:
```
root@OpenWrt:/# ethtool eth2 | grep -i speed
        Speed: 5000Mb/s
root@OpenWrt:/# iperf3 -c 192.168.1.199
Connecting to host 192.168.1.199, port 5201
[  5] local 192.168.1.1 port 48906 connected to 192.168.1.199 port 5201
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-1.00   sec   285 MBytes  2.39 Gbits/sec    9    673 KBytes       
[  5]   1.00-2.00   sec   296 MBytes  2.48 Gbits/sec    0    675 KBytes       
[  5]   2.00-3.00   sec   221 MBytes  1.84 Gbits/sec    0    675 KBytes       
[  5]   3.00-4.00   sec   287 MBytes  2.41 Gbits/sec    0    676 KBytes       
[  5]   4.00-5.00   sec   264 MBytes  2.22 Gbits/sec    0    677 KBytes       
[  5]   5.00-6.00   sec   271 MBytes  2.27 Gbits/sec    0    679 KBytes       
[  5]   6.00-7.00   sec   285 MBytes  2.39 Gbits/sec    0    679 KBytes       
[  5]   7.00-8.00   sec   275 MBytes  2.30 Gbits/sec    0    680 KBytes       
[  5]   8.00-9.00   sec   280 MBytes  2.35 Gbits/sec    0    682 KBytes       
[  5]   9.00-10.00  sec   267 MBytes  2.24 Gbits/sec    0    683 KBytes       
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec  2.67 GBytes  2.29 Gbits/sec    9             sender
[  5]   0.00-10.01  sec  2.67 GBytes  2.29 Gbits/sec                  receiver

iperf Done.
```
Server running on the PC and client on the r7800:
```
root@OpenWrt:/# ethtool eth2 | grep -i speed
        Speed: 5000Mb/s
root@OpenWrt:/# iperf3 -c 192.168.1.199 --bidir -i 10
Connecting to host 192.168.1.199, port 5201
[  5] local 192.168.1.1 port 35016 connected to 192.168.1.199 port 5201
[  7] local 192.168.1.1 port 35018 connected to 192.168.1.199 port 5201
[ ID][Role] Interval           Transfer     Bitrate         Retr  Cwnd
[  5][TX-C]   0.00-10.00  sec  1.38 GBytes  1.19 Gbits/sec    1   1.30 MBytes       
[  7][RX-C]   0.00-10.00  sec   287 MBytes   241 Mbits/sec                  
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID][Role] Interval           Transfer     Bitrate         Retr
[  5][TX-C]   0.00-10.00  sec  1.38 GBytes  1.19 Gbits/sec    1             sender
[  5][TX-C]   0.00-10.01  sec  1.38 GBytes  1.19 Gbits/sec                  receiver
[  7][RX-C]   0.00-10.00  sec   290 MBytes   243 Mbits/sec   39             sender
[  7][RX-C]   0.00-10.01  sec   287 MBytes   241 Mbits/sec                  receiver

iperf Done.

```
x86 Odroid H2+
```
iperf3 -c 192.168.1.1
Connecting to host 192.168.1.1, port 5201
[  5] local 192.168.1.199 port 44722 connected to 192.168.1.1 port 5201
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-1.00   sec   411 MBytes  3.45 Gbits/sec   45   1.24 MBytes       
[  5]   1.00-2.00   sec   412 MBytes  3.46 Gbits/sec    0   1.47 MBytes       
[  5]   2.00-3.00   sec   414 MBytes  3.47 Gbits/sec    0   1.66 MBytes       
[  5]   3.00-4.00   sec   412 MBytes  3.46 Gbits/sec    0   1.83 MBytes       
[  5]   4.00-5.00   sec   412 MBytes  3.46 Gbits/sec   45   1.46 MBytes       
[  5]   5.00-6.00   sec   414 MBytes  3.47 Gbits/sec    0   1.65 MBytes       
[  5]   6.00-7.00   sec   412 MBytes  3.46 Gbits/sec    0   1.83 MBytes       
[  5]   7.00-8.00   sec   414 MBytes  3.47 Gbits/sec    0   1.99 MBytes       
[  5]   8.00-9.00   sec   412 MBytes  3.46 Gbits/sec    0   2.14 MBytes       
[  5]   9.00-10.00  sec   412 MBytes  3.46 Gbits/sec    0   2.28 MBytes       
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec  4.03 GBytes  3.46 Gbits/sec   90             sender
[  5]   0.00-10.01  sec  4.03 GBytes  3.46 Gbits/sec                  receiver

iperf Done.
```

xh0823 commented:
可以发一个x86 64带无线和usb网卡的.config给我一份吗？

> 这增加了对
> 基于AQC111U控制器的USB到以太网Aquantia AQtion 5 / 2.5GbE适配器的支持。
> 
> 运行测试：x86
> 适配器测试：Sabrent NT-SS5G

lucize commented:
> I tried on mvebu - wrt3200 and is not working.

it does not compile or is not powering up, on either of the usb ports ?

mj22226 commented:
> it does not compile or is not powering up, on either of the usb ports ?

On mvebu it does compile,  power up, load the driver, the issue is that is not getting IP address.

```
[  118.582854] usb 3-1: new SuperSpeed Gen 1 USB device number 3 using xhci-hcd
[  118.628790] aqc111 3-1:1.0 eth1: register 'aqc111' at usb-f10f8000.usb3-1, Aquantia AQtion USB to 5GbE Controller, 00:24:26:55:03:55
[  118.843914] br-lan: port 5(eth1) entered blocking state
[  118.849173] br-lan: port 5(eth1) entered disabled state
[  118.854508] device eth1 entered promiscuous mode
[  118.859196] br-lan: port 5(eth1) entered blocking state
[  118.864444] br-lan: port 5(eth1) entered forwarding state
[  118.869984] br-lan: port 5(eth1) entered disabled state
[  177.294480] mv88e6085 f1072004.mdio-mii:00 lan4: Link is Up - 1Gbps/Full - flow control rx/tx
[  177.303064] br-lan: port 4(lan4) entered blocking state
[  177.308312] br-lan: port 4(lan4) entered forwarding state
[  177.314503] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
```

lucize commented:
> [  118.869984] br-lan: port 5(eth1) entered disabled state

did you tried it as a separate defined device ? are you sure you don't do a loop ? did you tried stp on br-lan ?

ynezz commented:
> I think you can remove the DEPENDS:=@TARGET_x86

And the x86 in the commit title, AFAIU there's nothing x86 specific.

mj22226 committed:
kernel: add Aquantia AQtion USB-to-5GbE adapters

This add support for USB-to-Ethernet Aquantia AQtion
5/2.5GbE adapters based on the AQC111U controllers.

Run-tested: x86
Adapter-tested: Sabrent NT-SS5G
Signed-off-by: Marty Jones <mj8263788@gmail.com>

ynezz commented:
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git

Neustradamus commented:
Merged commit is:
- https://github.com/openwrt/openwrt/commit/b63e57ba1795dfedd425d5cd6fffdae56274b3d1
